src/store.ts

```diff
@@ -57,5 +73,104 @@ export const createStore = () =>
           drags: { ...state.drags, [id]: newNode },
           nodes: [...state.nodes.filter(n => n.id !== id), newNode]
         };
+      }),
+    setCollapsedNodeIds: (nodeIds = []) =>
+      set(state => {
+        let collapsedNodeIds = [];
```

Comment:
This needs to be moved outside of this into its own method. If you can easily add a unit test, lets do that too.

Reply:
Moved this into its own utility file. Was going to make a unit test for it, but the repo is currently not set up for testing as is. Would you like me to get that all set up, or just ignore it?